src/org/armedbear/lisp/compile-file.lisp

```diff
@@ -880,18 +880,18 @@ COMPILE-FILE was invoked."
                          (*compiler-error-context* form))
                     (when (eq form in)
                       (return))
-                    (cond 
-                      ((>= (length (format nil "~a" form)) 65536)
-                       ;; Following the solution propose here:
-                       ;; see https://github.com/armedbear/abcl/issues/246#issuecomment-698854437
-                       ;; just include the offending interpreted form in the loader
-                       ;; using it instead of the compiled representation
-                       (diag "Falling back to interpreted version of top-level form longer ~
+
+                    (handler-case (process-toplevel-form form out nil)
+                      (compiler-bytecode-length-error ()
+                        ;; Following the solution propose here:
+                        ;; see https://github.com/armedbear/abcl/issues/246#issuecomment-698854437
+                        ;; just include the offending interpreted form in the loader
+                        ;; using it instead of the compiled representation
+                        (diag "Falling back to interpreted version of top-level form longer ~
```

Comment:
I'd like to retain the message to the `diag` channel.  This should be able to be suitably muffled…

Reply:
Oops.  Reading diffs too fast, sorry!